File: apps/frontend/src/services/buyer.service.ts
import { api } from './api';
import { Buyer } from '../features/buyers/types/buyers.types';

export const buyerService = {
  getBuyers(): Promise<Buyer[]> {
    return api.get<Buyer[]>('/buyers');
  },

  getBuyerById(id: string): Promise<Buyer> {
    return api.get<Buyer>(`/buyers/${id}`);
  },

  createBuyer(buyer: Omit<Buyer, 'id'>): Promise<Buyer> {
    return api.post<Buyer>('/buyers', buyer);
  },

  updateBuyer(id: string, buyer: Partial<Buyer>): Promise<Buyer> {
    return api.put<Buyer>(`/buyers/${id}`, buyer);
  },

  deleteBuyer(id: string): Promise<void> {
    return api.delete<void>(`/buyers/${id}`);
  },
};

export default buyerService;

File: apps/backend/src/main/java/com/goodearth/postsales/buyer/service/BuyerService.java
package com.goodearth.postsales.buyer.service;

import com.goodearth.postsales.buyer.dto.BuyerDto;
import java.util.List;

public interface BuyerService {
    List<BuyerDto> getAllBuyers();
    BuyerDto getBuyerById(java.util.UUID id);
    BuyerDto createBuyer(BuyerDto dto);
    BuyerDto updateBuyer(java.util.UUID id, BuyerDto dto);
    void deleteBuyer(java.util.UUID id);
}
